### DISCUSSION TITLE
Styles not being applied in docker environment local install.

### DISCUSSION THREAD

**Discussion by arielcorte** [chatwoot/chatwoot#8122] [Contributors]
### Describe the bug

I've followed the guide in [docker environment setup](https://www.chatwoot.com/docs/contributing-guide/environment-setup/docker).

First it failed because of some permission denied error, fixed by using command `docker compose build --no-cache base` or by using `master` branch

Then when it finally worked, I go to `localhost:3000` and I see this:
![image](https://github.com/chatwoot/chatwoot/assets/86988649/a99cd29f-434d-4bc7-a2c5-e5f52361c998)
The app seems to work, I can login with the default dev credentials, but cant get the app to apply the styles.

I tried then running `make burn` (following [project-setup](https://www.chatwoot.com/docs/contributing-guide/project-setup/#docker-for-development)) but that brings a lot of other problems such as the pg, which I couldn't fix with [the common error page](https://www.chatwoot.com/docs/contributing-guide/common-errors#pg-gem-installation-error), or a permission error. Also I don't think it is the path to follow since I'm using docker and I shouldn't have to install something locally.

The line in the `docker compose up` logs that give me a smell is 
```
chatwoot-rails-1     | + bundle install
yarn install v1.22.19
[1/5] Validating package.json...
error @chatwoot/chatwoot@3.1.1: The engine "node" is incompatible with this module. Expected version "20.x". Got "18.18.2"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
chatwoot-webpack-1 exited with code 1

```
I tried entering the chatwoot:development container shell, run `node -v` and it returns `18.x.x`. I update nodejs-current manually from shell to version `20.x.x` and nothing fixes.

Maybe I'm getting something wrong cause I couldn't find any other issue with this error other than [this one](https://github.com/chatwoot/chatwoot/issues/7928), which is in self hosted linux and not in docker.

### To Reproduce

1. Copy the repo
2. add a password to redis and postgres in `.env` file and the same password to `docker-compose.yaml`.
3. run `sudo docker compose build --no-cache base`.
4. run `sudo docker compose run --rm rails bundle exec rails db:chatwoot_prepare`.
5. run `sudo docker compose up` (even if the containers were running).
6. go to `localhost:3000` and it shows the login app all broken up.

### Expected behavior

To run the app via docker without any errors and show exactly the same as if it was hosted in another platform.

### Environment

Docker

### Cloud Provider

None

### Platform

None

### Operating system

Fedora 38

### Browser and version

Google Chrome 117.0.5938.149

### Docker (if applicable)

- `docker version`
Client: Docker Engine - Community
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:33:40 2023
 OS/Arch:           linux/amd64
 Context:           default

- `docker info`
Client: Docker Engine - Community
 Version:    24.0.6
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.21.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

- `docker compose version`
Docker Compose version v2.21.0

**Comment by pranavrajs** [comment]
I am moving this to a discussion as there is no confirmed bug at the moment. @vishnu-narayanan Can you please help here?

**Comment by arielcorte** [comment]
Are the docs at [environment-setup docker](https://www.chatwoot.com/docs/contributing-guide/environment-setup/docker) updated? Because in the docker-compose file it looks like the file pulls from `chatwoot:development`, which doesn't seem to exist in dockerhub (just `chatwoot/chatwoot:develop` exists)

**Comment by shomid** [comment]
I have the same problem

**Comment by jyotman** [comment]
@ArielCorte were you able to fix this?
I'm facing a similar issue. I'm following the same guide and trying to run on macbook M2 using docker - https://www.chatwoot.com/docs/contributing-guide/environment-setup/docker

I can see that the webpack container crashes with the error - "yarn not found" and I believe that's the reason styles are not getting applied.
cc: @pranavrajs

**Reply by arielcorte** [reply]
Nope, didn't fix it and moved to another project. I forgot most of it.
